File: ch1/1-23.go
package main

import (
	"bufio"
	"fmt"
	"os"
)

const maxLine = 1000

func main() {
	for {
		line, l := getLine(maxLine)
		var ignore bool
		for i := 0; i < l; i++ {
			if rune(line[i]) == '/' && i+1 < l && rune(line[i+1]) == '/' {
				ignore = true
			}
			if !ignore {
				fmt.Printf("%c", rune(line[i]))
			}
		}
		if ignore {
			fmt.Printf("\n")
		}
	}
}

func getLine(lim int) ([]byte, int) {
	var c byte
	var i int
	s := make([]byte, lim)
	r := bufio.NewReader(os.Stdin)
	for i = 0; i < lim-1; i++ {
		c, err := r.ReadByte()
		if rune(c) == '\n' || err != nil {
			break
		}
		s[i] = c
	}
	if rune(c) == '\n' {
		s[i] = c
		i++
	}
	return s, i
}

File: ch1/1-23.c
#include <stdio.h>

#define MAXLINE 1000

int get_line(char s[], int lim);

int main(void)
{
    char line[MAXLINE];
    int len;
    while ((len = get_line(line, MAXLINE)) > 0) {
        int ignore = 0;
        for (int i = 0; i < len; i++) {
            if (line[i] == '/' && i+1 < len && line[i+1] == '/') {
                ignore = 1;
            }
            if (!ignore) {
                putchar(line[i]);
            }
        }
        if (ignore) {
            printf("\n");
        }
    }
    return 0;
}

int get_line(char s[], int lim)
{
    int c, i;
    for (i = 0; i < lim - 1 && (c = getchar()) != EOF && c != '\n'; i++) {
        s[i] = c;
    }
    if (c == '\n') {
        s[i] = c;
        ++i;
    }
    s[i] = '\0';

    return i;
}
